src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,165 @@
+package duke.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.IncorrectCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+
+/**
+ * Represents the Parser Class.
+ * Responsible for parsing user input.
+ *
+ * @author Shishir
+ */
+public class Parser {
+
+    /**
+     * Returns a command based on user input.
+     * @param fullCommand String representation of user input.
+     * @return Command based on user input.
+     */
+    public static Command parse(String fullCommand) {
+
+        String[] split = fullCommand.split(" ", 2);
+        Command c = null;
+
+        switch(split[0]) {
+        case "bye":
+            c = validateCommand(split, true);
+            break;
+        case "list":
+            c = validateCommand(split, false);
+            break;
+        case "mark":
+        case "unmark":
+            c = validateIndex(split, true);
+            break;
+        case "todo":
+        case "deadline":
+        case "event":
+            c = validateTask(split);
+            break;
+        case "delete":
+            c = validateIndex(split, false);
+            break;
+        case "find":
+            c = validateFind(split);
+            break;
+        default:
+            c = new IncorrectCommand("I'm sorry, I couldn't understand that. Please try again!");
+        }
+        return c;
+    }
+
```

Comment:
I think having all of the validation checks in one section is pretty confusing, especially when most of them are exclusive to one command! Perhaps use parser to handle the input into a form readable by each command, and then handle the validation there?